docs/advanced/classes.md

```diff
@@ -0,0 +1,547 @@
+## Classes
+
+Class describes the common properties and methods of the created object.
+
+### Supported Patterns
+
+```yaml
+name: Advanced classes
+```
+
+#### Semantic: Class
+
+##### Examples
+
+###### Call + Assigned call
+A class is instantiated and we assign one of its functions to a variable and then call that variable.
+
+```ts
+class MyClass {
+    function func() {
+        /*Empty*/
+    }
+}
+
+const a = new MyClass();
+const b = a.func();
+b();
+
+```
+
+```yaml
+relation:
+    type: call
+    implicit: true
+    items:
+        -   from: file:'<File file0.ts>'
+            to: method:'Myclass.fuc'
+            loc: 8:1:1
+```
+
+###### Assigned self call
+A class is instantiated and we call one of its functions. The function called
+assigns `self` to a variable and using that variable we call a different
+function contained in the class.
+```ts
+class MyClass {
+    function func1() {
+        /* Empty */
+    }
+
+    function func2() {
+        const a = this;
+        a.func1();
+    }
+}
+
+const a = new MyClass();
+a.func2();
+
+```
+
+```yaml
+relation:
+    type: call
+    implicit: true
+    items:
+        -   from: file:'<File file1.ts>'
+            to: method:'MyClass.func2'
+            loc: 12:1:1
+        -   from: method:'MyClass.func2'
+            to: method:'MyClass.func1'
+            loc: 12:3:5
+```
+
+###### Base class calls child
+
+```ts
+class A {
+    func(){
+        this.child();
+    }
+
+    child() {
+        /* Empty */
+    }
+}
+
+class B extends A {
+    constructor() {
+        super();
+        this.child = this.func2;
+    }
+
+    func2(){
+       /* Empty */
+    }
+}
+
+class C extends A {
+    constructor() {
+        super();
+        this.child = this.func2;
+    }
+
+    func2(){
+        /* Empty */
+    }
+}
+
+const b = new B();
+b.func();
+
+const c = new C();
+c.func();
+
+```
+
+```yaml
+relation:
+    items:
+        -   from: class:'B'
+            to: class:'A'
+            loc: file2:13:17
+            type: extend
+        -   from: class:'C'
+            to: class:'A'
+            loc: file2:22:17
+            type: extend
+        -   from: method:'B.func'
+            to: method:'B.func2'
+            loc: 34:1:6
+            type: call
+            implicit: true
+        -   from: method:'C.func'
+            to: method:'C.func2'
+            loc: 36:1:6
+            type: call
+            implicit: true
+```
+###### Direct call
+```ts
+class MyClass {
+    func(){
+        /*Empty*/
+    }
+}
+
+new MyClass().func();
+
+```
+```yaml
+relation:
+    type: call
+    extra: false
+    implicit: false
+    items:
+        -   from: file:'<File file3.ts>'
+            to: method:'Myclass.func'
+            loc: 7:5:9 
+```
+###### Import attr access
+```ts
+Function func(){
+    /*Empty*/
+}
+export { func };
+```
+```ts
+import * as to_import from './file4.ts'; 
```

Comment:
Style: While migrating PyCG's dataset, please also change the variable naming flavor to JS's `camelCase` style, not using `_`.